application/sbsk-web/src/main/java/com/sbsk/web/logging/appender/FluentAppender.java

```diff
@@ -90,32 +109,57 @@ public void stop() {
 
   @Override
   protected void append(ILoggingEvent event) {
-    Map<String, Object> messages = new HashMap<String, Object>();
-
-    messages.put("host", getHostname());
-    messages.put("app", label);
-    messages.put("main", System.getProperty("sun.java.command"));
-    messages.put("thread", event.getThreadName());
-    messages.put("level", event.getLevel().toString());
-    messages.put("class", event.getLoggerName());
-    if (event.getThrowableProxy() == null) { // Without exception
-      messages.put("exception", "--- This.log.is.not.an.exception ---");
-      Throwable tempThrowable = new Throwable();
-      tempThrowable.setStackTrace(event.getCallerData());
-      messages.put("message", event.getFormattedMessage().toString());
-      if (event.getCallerData() != null) {
-        messages.put("stacktrace", extractCallerData(tempThrowable));
-      }
-    } else { // With exception
-      ThrowableProxy throwableProxy = (ThrowableProxy) event.getThrowableProxy();
-      messages.put("exception", throwableProxy.getClassName());
-      messages.put("message", throwableProxy.getMessage());
-      messages.put("stacktrace", extractCallerData(throwableProxy.getThrowable()));
+    Map<String, Object> messages = new HashMap<>();
+
+    addBaseData(event, messages);
+    if (eventHasExceptions(event)) {
+      addExceptionData(event, messages);
+    } else {
+      addMessageData(event, messages);
```

Comment:
you could use Optional here, something like that:

Optional.of(event.getThrowableProxy()).ifPresent()
 .then(addExceptionData(event, messages))
 .otherwise(addMessageData(event, messages))